.github/workflows/spark-integration-test.yml

```diff
@@ -108,3 +107,64 @@ jobs:
             spark-connector/v3.5/spark/build/spark-3.5-integration-test.log
             distribution/package/logs/*.out
             distribution/package/logs/*.log
+
+  SparkIT-on-pr:
+    needs: changes
+    if: (github.event_name == 'pull_request' && needs.changes.outputs.source_changes == 'true')
```

Comment:
why you split event to `push` and `pull_request`, what's the difference between the two jobs except java version?

Reply:
To decrease the number of SparkCI jobs. Since the 'pull_request' is more frequent than 'push', I reduced the multi-jdk running tests under 'pull_request' and moved them to be tested after PR merging to alleviate CI pressure.

Except for the Java version, everything else is the same.

Reply:
Is there any way to reuse the code?  duplicated code are hard to maintance.

Reply:
I can't find a good way to reuse the code. It requires importing a new file called `spark-common-steps.yml`, which will make things more complex.

Reply:
> Is there any way to reuse the code? duplicated code are hard to maintance.

+1. The current situation seems quite complicated to me and may introduce excessive time to maintain so many CI trigger condition.

Reply:
Since two people have suggested doing this, I'll try to extract the public part

Reply:
@FANNG1 @yuqi1129 updated, plz help review again, thanks!